cargo/lib/dependabot/cargo/file_updater/manifest_updater.rb

```diff
@@ -100,8 +100,9 @@ def update_manifest_req(content:, dep:, old_req:, new_req:)
           if simple_declaration
             simple_declaration_regex =
               /(?:^|["'])#{Regexp.escape(simple_declaration)}/
+            old_req_escped = Regexp.escape(old_req)
             content.gsub(simple_declaration_regex) do |line|
-              line.gsub(/.+=.*\K(#{old_req})/, new_req)
+              line.gsub(/.+=.*\K(#{old_req_escped})/, new_req)
```

Comment:
```suggestion
              line.gsub(/.+=.*\K(#{old_req_escaped})/, new_req)
```

Typo. It wouldn't let me do a multi-line suggestion sorry